src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BuildTest.cs

```diff
@@ -4072,5 +4072,20 @@ public void PackageNamingPolicy ([Values ("LowercaseMD5", "LowercaseCrc64")] str
 				Assert.AreEqual ($"__XA_PACKAGE_NAMING_POLICY__={packageNamingPolicy}", File.ReadAllText (environment).Trim ());
 			}
 		}
+
+		[Test]
+		public void ValidateFIPSPolicy ()
+		{
+			if (!IsWindows) {
+				Assert.Ignore ("Ignoring, this test is only valid on Windows.");
+			}
+
+			int fipsEnabled = (int) Microsoft.Win32.Registry.GetValue (@"HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Lsa\FipsAlgorithmPolicy", "Enabled", 0);
+			if (fipsEnabled == 0) {
+				Assert.Ignore ("Ignoring, FIPS is not enabled.");
+			}
+
+			Assert.Throws<TargetInvocationException> (() => System.Security.Cryptography.MD5.Create ());
+		}
```

Comment:
We might not even need this test, but it will validate the powershell step is working.